onmt/trainer.py

```diff
@@ -140,9 +150,13 @@ def train(self, train_iter_fct, valid_iter_fct, train_steps, valid_steps):
 
                 norm = self._norm(batch)
 
-                use_tf = random.random() < self._teacher_forcing_ratio
+                batch_teacher_forcing_ratio = self._calc_teacher_forcing_ratio(step, 
+                                                self._sampling_type,
+                                                self._scheduled_sampling_decay, 
+                                                self._scheduled_sampling_k,
+                                                self._scheduled_sampling_c)
```

Comment:
I don't think wee need to pass these attributes as arguments of the method `self._calc_teacher_forcing_ratio` because the method already has access to them through `self`.